python/pyarrow/_fs.pyx

```diff
@@ -260,6 +349,48 @@ cdef class FileSelector(_Weakrefable):
         If true, an empty selection is returned.
     recursive : bool, default False
         Whether to recurse into subdirectories.
+
+    Examples
+    --------
+    Generate a file:
+
+    >>> from pyarrow import fs
+    >>> local = fs.LocalFileSystem()
+    >>> path_s = local_path + '/pyarrow-s-example.dat'
+    >>> with local.open_output_stream(path_s) as stream:
+    ...     stream.write(b'data')
+    4
+
+    Create new directory and subdirectory with copied data:
+
+    >>> # directory
+    >>> local.create_dir(local_path + '/alphabet')
+    >>> local.copy_file(path,
+    ...                 local_path + '/alphabet/pyarrow-s-example.dat')
+    >>> # subdirectory
+    >>> local.create_dir(local_path + '/alphabet/subdir')
+    >>> local.copy_file(path,
+    ...                 local_path + '/alphabet/subdir/pyarrow-s-example.dat')
```

Comment:
Is all this setup code inside the example useful?
How about making the example non-executable and just show some cooked up `get_file_info` results?

Reply:
Sure, can do 👍